src/asm/random.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// Copyright (c) 2022 Amazon.com, Inc. or its affiliates.
+//
+// Author(s):
+//   - Ali Saidi <alisaidi@amazon.com>
+
+#[derive(Copy, Clone, Debug)]
```

Comment:
Could you put this before the `pub struct`? It's easy to miss otherwise due to the big comment.